### DISCUSSION TITLE
Should there be a warning when using `peerDependencies` rather than `dependencies` in such a case?

### DISCUSSION THREAD

**Discussion by aleen42** [npm/feedback#553] [CLI]
I noticed that `@babel/preset-env` has specified its `peerDependencies` with `@babel/core` and found that `npm@7.x` would automatically install it without any specification of `@babel/core` again, but `npm@6.x` cannot.

So I want to know that whether `npm` can give out a warning when using `peerDependencies` that low version npm won't install it automatically, which means that you need to specify it yourself.

**Comment by ljharb** [comment]
npm 6 would have to add such a warning - there's no point in npm 7 outputting it, since it'd be showing it to *consumers* of the package and not the author.

In npm 3-6, it was always the consumer's responsibility to run `npm ls` - which, if it exits nonzero, means the entire dependency graph is invalid, and nothing can be expected to work.

This is still the case in npm 7; it's just that npm 7's install process will more proactively warn you before installing an invalid dep graph.

**Reply by aleen42** [reply]
To the author: npm 7 can notice that if specifying `peerDependencies`, your library may need consumers to specify in their own package.
To the consumer: npm 6 or loser can notice that because the author has specified `peerDependencies`, you may need to specify it in your own package because npm won't automatically install it for you.

**Reply by ljharb** [reply]
npm < 6 is EOL, and npm 6 is only getting security updates.

npm 7 would have to warn on `install` of the package. I'd say you could file an RFC for that, but it won't solve the underlying problem - which is that if you're using npm 3-6, you *have always* had to Just Know to run `npm ls` and check the validity of your graph manually.

**Reply by aleen42** [reply]
I know it.
